extensions/amp-a4a/0.1/a4a-variable-source.js

```diff
@@ -113,10 +112,66 @@ export class A4AVariableSource extends VariableSource {
       return getNavigationData(this.win_, 'redirectCount');
     });
 
+    this.set('HTML_ATTR',
+        /** @type {function(...*)} */(this.htmlAttrBinding_.bind(this)));
+
     for (let v = 0; v < WHITELISTED_VARIABLES.length; v++) {
       const varName = WHITELISTED_VARIABLES[v];
       const resolvers = this.globalVariableSource_.get(varName);
       this.set(varName, resolvers.sync).setAsync(varName, resolvers.async);
     }
   }
+
+  /**
+   * Provides a binding for getting attributes from the DOM.
+   * Most such bindings are provided in src/service/url-replacements-impl, but
+   * this one needs access to this.win_.document, which if the amp-analytics
+   * tag is contained within an amp-ad tag will NOT be the parent/publisher
+   * page. Hence the need to put it here.
+   * @param {string} cssSelector Elements matching this selector will be
+   *     included, provided they have at least one of the attributeNames
+   *     set, up to a max of 10. May be URI encoded.
+   * Note: Additional params will be the names of the attributes whose values
+   * will be returned. There should be at least 1.
+   * @returns {string} A stringified JSON array containing one member for each
+   *     matching element. Each member will contain the names and values of the
+   *     specified attributes, if the corresponding element has that attribute.
+   *     Note that if an element matches the cssSelected but has none of the
+   *     requested attributes, then nothing will be included in the array
+   *     for that element.
+   */
+  htmlAttrBinding_(cssSelector) {
+    const HTML_ATTR_MAX_RETURN_SIZE = 10;
+    const attributeNames = Array.prototype.slice.call(arguments, 1);
+    if (!cssSelector || !attributeNames.length) {
+      return '[]';
+    }
+    cssSelector = decodeURI(cssSelector);
+    let elements;
+    try {
+      elements = this.win_.document.querySelectorAll(cssSelector);
```

Comment:
You could just move all of the code beyond this into the try/catch and then you could do:

try {
    this.win_.document.querySelectorAll(cssSelector).forEach(...
} catch ...

Reply:
Had that originally, Jeff said to switch to this.
https://github.com/ampproject/amphtml/pull/13712#discussion_r171349021